services/dedup/dedup.go

```diff
@@ -192,59 +202,67 @@ func (d *DedupHandleT) writeToBadger(messageIDs []string) error {
 
 // MarkProcessed persist messageIDs in Disk, with expiry time of dedupWindow
 // Any message mark here will appear in FindDuplicates() if queried inside the dedupWindow
-func (d *DedupHandleT) MarkProcessed(messageIDs []string) error {
-	return d.writeToBadger(messageIDs)
+func (d *Manager) MarkProcessed(infos []Info) error {
+	return d.writeToBadger(infos)
 }
 
-func (d *DedupHandleT) FindDuplicates(messageIDs []string, allMessageIDsSet map[string]struct{}) (duplicateIndexes []int) {
-	toRemoveMessageIndexesSet := make(map[int]struct{})
+func (d *Manager) FindDuplicates(infos []Info, allMessageIDsSet map[string]Payload) map[int]Payload {
+	toRemoveMessageIndexesSet := make(map[int]Payload)
 	// Dedup within events batch in a web request
 	messageIDSet := make(map[string]struct{})
 
 	// Eg messageIDs: [m1, m2, m3, m1, m1, m1]
 	// Constructing a set out of messageIDs
-	for _, messageID := range messageIDs {
-		messageIDSet[messageID] = struct{}{}
+	for _, info := range infos {
+		messageIDSet[info.MessageID] = struct{}{}
 	}
 	// Eg messageIDSet: [m1, m2, m3]
 	// In this loop it will remove from set for first occurrence and if not found in set it means it's a duplicate
-	for idx, messageID := range messageIDs {
-		if _, ok := messageIDSet[messageID]; ok {
-			delete(messageIDSet, messageID)
+	for idx, info := range infos {
+		if _, ok := messageIDSet[info.MessageID]; ok {
+			delete(messageIDSet, info.MessageID)
 		} else {
-			toRemoveMessageIndexesSet[idx] = struct{}{}
+			toRemoveMessageIndexesSet[idx] = Payload{}
 		}
 	}
 	// Dedup within batch of batch jobs
-	for idx, messageID := range messageIDs {
-		if _, ok := allMessageIDsSet[messageID]; ok {
-			toRemoveMessageIndexesSet[idx] = struct{}{}
+	for idx, info := range infos {
+		if _, ok := allMessageIDsSet[info.MessageID]; ok {
+			toRemoveMessageIndexesSet[idx] = Payload{}
 		}
 	}
 
 	// Dedup with badgerDB
 	err := d.badgerDB.View(func(txn *badger.Txn) error {
-		for idx, messageID := range messageIDs {
-			_, err := txn.Get([]byte(messageID))
-			if err != badger.ErrKeyNotFound {
-				toRemoveMessageIndexesSet[idx] = struct{}{}
+		for idx, info := range infos {
+			item, err := txn.Get([]byte(info.MessageID))
+			if err != nil && err != badger.ErrKeyNotFound {
+				pkgLogger.Warn("[[ Dedup ]] Error while getting item from badgerDB", err)
```

Comment:
Adding a warning log instead of returning it. Right now, we are not aware of the scale at which we are getting this error. Since the error is being handled as panic.